modules/fdb/examples/sqlite-to-fdb.rs

```diff
@@ -0,0 +1,155 @@
+                "select count(distinct [{}]) as unique_count from {}",
+                template_table.column_at(0).unwrap().name(),
+                template_table.name()
+            ),
+            rusqlite::NO_PARAMS,
+            |row| row.get(0),
+        )?;
+
+        // Bucket count should be 0 or a power of two
+        let new_bucket_count = if unique_key_count == 0 {
+            0
+        } else {
+            u32::next_power_of_two(unique_key_count)
+        };
+
+        // Create destination table
+        let mut dest_table = store::Table::new(new_bucket_count as usize);
+
+        // Number of columns destination table should have
+        let column_count = template_table.column_count();
+
+        // Vector to store target datatypes in as these can't be determined from the sqlite source db
+        let mut target_types: Vec<ValueType> = Vec::with_capacity(column_count);
+
+        // Build select query with the same column names and order as the template FDB
+        let mut select_query = String::from("select ");
+
+        // Write select query and store target datatypes
+        for index in 0..column_count {
+            let template_column = template_table
+                .column_at(index).unwrap();
+
+            let template_column_name = template_column.name();
+
+            write!(select_query, "[{}]", template_column_name).unwrap();
+
+            if index < column_count - 1 {
+                write!(select_query, ", ").unwrap();
+            }
+
+            target_types.push(template_column.value_type());
+
+            dest_table.push_column(template_column.name_raw(), template_column.value_type());
+        }
+
+        select_query.push_str(" from ");
+        select_query.push_str(&template_table.name());
+
+        // Execute query
+        let mut statement = conn.prepare(&select_query)?;
+        let mut rows = statement.query(rusqlite::NO_PARAMS)?;
+
+        // Iterate over rows
+        while let Some(sqlite_row) = rows.next()? {
+            let mut fields: Vec<Field> = Vec::with_capacity(column_count);
+
+            // Iterate over fields
+            for index in 0..column_count {
+                fields.push(match sqlite_row.get_raw(index) {
+                    ValueRef::Null => Field::Nothing,
+                    ValueRef::Integer(i) => match target_types[index] {
+                        ValueType::Integer => Field::Integer(i as i32),
+                        ValueType::Boolean => Field::Boolean(i == 1),
+                        ValueType::BigInt => Field::BigInt(i),
+                        _ => panic!("Invalid target datatype; cannot store SQLite Integer as FDB {:?}", target_types[index]),
+                    },
+                    ValueRef::Real(f) => Field::Float(f as f32),
+                    ValueRef::Text(t) => match target_types[index] {
+                        ValueType::Text => {
+                            Field::Text(String::from(std::str::from_utf8(t).unwrap()))
```

Comment:
You can probably replace a lot of `unwrap()` with the `?` operator i.e.
```suggestion
                            Field::Text(String::from(std::str::from_utf8(t)?))
```

Reply:
Changed `.unwrap()` to `?` on the functions that return a `Result`, there are two usages left on `Option`s (line 50 and 78) but as I understand it these can't be replaced in the same way because `main` returns a `Result`.
I believe it is acceptable to use `.unwrap()` on those, because line 50 tries to get the first column of a table which will always be there as SQLite doesn't support zero-column tables, and line 78 only attempts to read columns at indices in `0..column_count`, so those columns should always be there.

Reply:
I guess the idea of preferring `?` over `unwrap` here is what happens when someone runs this tool with a broken FDB file as a template anyways. The panic may be confusing, so it's really fine in practice, but maybe helpful for those holding it wrong.

From a practical point of view, you can use [`Option::ok_or_else`](https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.ok_or_else) to turn that option into a result. For the second case, you could also simply use [`Table::column_iter`](https://docs.rs/assembly-data/0.3.0/assembly_data/fdb/mem/struct.Table.html#method.column_iter) with [`Iterator::enumerate`](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.enumerate) to avoid the lookup by ID. None of this is a requirement though.

Reply:
Thanks for both suggestions, I've implemented them!